File: 198-HouseRobber/198-HouseRobber.ts
// Last updated: 16/7/2026, 9:20:05 pm
function rob(nums: number[]): number {
    const ans: number[] = [];
    ans.push(0);
    ans.push(nums[0]);
    for (let i = 2; i <= nums.length; i++) {
        const a= ans[i - 1];
        const b= ans[i - 2] + nums[i-1];
        ans.push(Math.max(a,b));
    }
    return ans[nums.length];
};
rob([1,2,3,1])

File: 198-HouseRobber/198-HouseRobber.py
# Last updated: 16/7/2026, 9:19:59 pm
class Solution:
    def rob(self, nums: List[int]) -> int:
        ans:List[int] = []
        ans.append(0)
        ans.append(nums[0])
        for i in range(1,len(nums)):
            a = ans[i]
            b = nums[i] + ans[i-1]
            ans.append(max(a,b))
        return ans[-1] 
        